### ISSUE TITLE
OAuth social login fails

### ISSUE THREAD

**Issue by yoshiokatsuneo** [angular-fullstack/generator-angular-fullstack#1390]
OAuth social login fails on 3.0.0-rc8 .
On mongoose 4.x, model.save()'s callback have two arguments: an object and a number.
So, at the line "user.saveAsync().then(function(user)" of "server/auth/{facebook,google,twitter}/passport.js", the "user" argument is not object itself but an array "[object, 1]".
Changing from "return done(null, user);" to "return done(null, user[0]);" fixes the error.

(app/templates/server/auth/{facebook,google,twitter}/passport.js)
```
          user.saveAsync()
            .then(function(user) {
-              return done(null, user);
+             return done(null, user[0]);
            })
```

**Comment by yoshiokatsuneo** [comment]
I made the pull request.
https://github.com/DaftMonk/generator-angular-fullstack/pull/1392

**Comment by smirciat** [comment]
I am facing a similar issue, but this fix does not help me for some reason.  Running 3.0.0-rc4, each of the 3 generated oAuth methods (Facebook, Google, Twitter) fails with "invalid token."  Each is getting the token returned to the server but fails on jwt verification.  Same setup with 2.1.1 works perfectly.  Could be a separate issue?

**Comment by yoshiokatsuneo** [comment]
@smirciat How about using the latest version: 3.0.0-rc8 ?

**Comment by smirciat** [comment]
@yoshiokatsuneo That worked.  Any ideas what might have changed in oAuth between rc4 and rc8?  Side by side the code looks the same to me.  Looking to update an existing project that was initially generated from 3.0.0-rc4

**Comment by MichaelLeeHobbs** [comment]
Issue title should be "OAuth social login fails on new user login" would be easier to find for those of us searching for it. Not sure if issues are renameable.

**Comment by Awk34** [comment]
I believe this has been fixed.

**Comment by yoshiokatsuneo** [comment]
Actually, it looks not fixed, yet.

**Comment by Awk34** [comment]
Scaffold out a new app with 3.4.0 and see if things don't act as expected

**Comment by yoshiokatsuneo** [comment]
@Awk34 Ah, I'm sorry but I missed. Now, I confirmed that the issue is fixed on the latest 3.4.0. Thanks !
